### PR TITLE
Nvar visitors

### PR DESCRIPTION
**Pull request by JulienVdG**
This goes on top of #240 It implements table, extract, assemble, parsedir visitors for NVARs (and store)

### CODE REVIEW

File: pkg/uefi/nvram.go [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+			}
+		}
+		if calculatedChecksum != 0 {
+			calculatedChecksum = -calculatedChecksum
+			v.ExpectedChecksum = &calculatedChecksum
+		}
+		knownExtDataFormat = true
+	}
+
+	if v.Header.Attributes&NVarEntryAuthWrite == 0 {
```
Comment (didjcodt) [inline pkg/uefi/nvram.go]: It seems the `v.Header.Attributes` is of type `NVarAttribute`. Is this line different from `v.Header.Attributes == NVarEntryAuthWrite`?

Comment (JulienVdG) [reply pkg/uefi/nvram.go]: NVarAttribute is actually a bitfield so here we check if the bit 6 (NVarEntryAuthWrite) is not set

Comment (didjcodt) [reply pkg/uefi/nvram.go]: Oh ok sorry, I didn't understand that.

File: pkg/uefi/nvram.go [hunk 2]
```diff
@@ -112,17 +113,57 @@ type NVar struct {
...
+}
+
+// Apply calls the visitor on the NVar.
+func (v *NVar) Apply(vr Visitor) error {
+	return vr.Visit(v)
+}
+
+// ApplyChildren calls the visitor on each child node of NVar.
+func (v *NVar) ApplyChildren(vr Visitor) error {
+	if v.NVarStore != nil {
```
[the remark this answers is not in the document]
Comment (JulienVdG) [reply pkg/uefi/nvram.go]: For posterity as we clarified this IRL:
Yes NVar can contain an NVarStore.
Typically a Variable named like "defaults" whould contain a store with a few variables in.

[the remark this answers is not in the document]
Comment (JulienVdG) [reply pkg/uefi/nvram.go]: IE: This is not a ref to the parent, no risk of infinite recursion.

[the remark this answers is not in the document]
Comment (GanShun) [reply pkg/uefi/nvram.go]: Thanks!

**Pull request merged**